tests/integration/seccomp-notify.bats

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function setup() {
+	teardown_seccompagent
+	setup_seccompagent
+
+	setup_busybox
+}
+
+function teardown() {
+	teardown_seccompagent
+	teardown_bundle
+}
+
+# Create config.json template with SCMP_ACT_NOTIFY actions
+# $1: command to run
+# $2: noNewPrivileges (false/true)
+# $3: list of syscalls
+function scmp_act_notify_template() {
+	# The agent intercepts mkdir syscalls and creates the folder appending
+	# "-bar" (listenerMetadata below) to the name.
+	update_config '.process.args = ["/bin/sh", "-c", "'"$1"'"] |
+	.process.noNewPrivileges = '"$2"' |
+	.linux.seccomp = {
+		"defaultAction":"SCMP_ACT_ALLOW",
+		"listenerPath": "'"$SECCCOMP_AGENT_SOCKET"'",
+		"listenerMetadata": "bar",
+		"architectures":["SCMP_ARCH_X86","SCMP_ARCH_X32", "SCMP_ARCH_X86_64" ],
+		"syscalls":[{"names":['"$3"'], "action":"SCMP_ACT_NOTIFY"}]
+	}'
+}
+
+# Support for seccomp notify requires Linux > 5.6 because
+# runc uses the pidfd_getfd system call to fetch the seccomp fd.
+# https://github.com/torvalds/linux/commit/8649c322f75c96e7ced2fec201e123b2b073bf09
+
+# The call to seccomp is done at different places according to the value of
+# noNewPrivileges, for this reason many of the following cases are tested with
+# both values.
+
+# check that SCMP_ACT_NOTIFY on older kernels fails with a clear error message.
+@test "runc run [seccomp] (SCMP_ACT_NOTIFY old kernel)" {
+	requires arch_x86_64
+
+	if [[ "$KERNEL_MAJOR" -gt 5 || ("$KERNEL_MAJOR" -eq 5 && "$KERNEL_MINOR" -ge 6) ]]; then
+		skip "requires kernel less than 5.6"
+	fi
+
+	scmp_act_notify_template "/bin/true" false '"mkdir"'
+
+	runc run test_busybox
+	[ "$status" -ne 0 ]
+}
+
+@test "runc run [seccomp] (SCMP_ACT_NOTIFY noNewPrivileges false)" {
+	requires_kernel 5.6
+	requires arch_x86_64
```

Comment:
If these are requirements for every tests (it seems so), then it's better to move this to `setup()` (so we can skip setup entirely).

Reply:
Only the first test doesn't require this. I've moved this to the `setup()` function and created a new file for the first file (tests that we give a clear error msg on older kernels). Let me know what you think